node/src/braid/mod.rs

```diff
@@ -248,7 +254,70 @@ impl Braid {
             }
         }
     }
+    /// utility function for GetBeadsAfter request
+    pub fn get_beads_after(&self, old_tips: Vec<BeadHash>) -> Option<Vec<Bead>> {
+        let old_tips: HashSet<BeadHash> = old_tips.into_iter().collect();
+        tracing::warn!(
+            old_tips=?old_tips,"Tips received from the peer for which beads are requested for during IBD"
+        );
+        //In case no tips are present i.e. the new braid-node has been initialized
+        if old_tips.len() == 0 {
+            return Some(self.beads.clone());
+        }
+        let mut response_beads = Vec::new();
+        let mut found_start = false;
+        let mut smallest_index = usize::MAX;
+        //finding the starting index
+        for hash in &old_tips {
+            if let Some(&index) = self.bead_index_mapping.get(hash) {
+                if index < smallest_index {
+                    smallest_index = index;
+                }
+                found_start = true;
+            }
+        }
+        //If somehow no bead matched that can be due to possible latency/fork so send all the beads instead as fallback
+        if smallest_index == usize::MAX {
+            return Some(self.beads.clone());
+        }
+
+        tracing::debug!(
+            smallest_index=?smallest_index,"Smallest possible index from all the tips - ",
+
+        );
+        // just iterating over the vector of cohorts for now, this needs to be changed to use a more efficient retrieval of cohort index given bead hash
+        let mut smallest_cohort_index = usize::MAX;
+        for (idx, cohort) in self.cohorts.iter().enumerate() {
```

Comment:
Hmmm. This is bad.

Add a reverse mapping `HashMap<bead_idx: usize, cohort_idx: usize>` in Braid so we can use it here.